src/package.ts

```diff
@@ -33,32 +33,29 @@ export function up(options?: find.Options): string | undefined {
  */
 export function cache(
 	name: string,
-	options?: find.Options & { create?: boolean },
+	options?: FindOptions & { create?: boolean },
 ): string | undefined {
-	options ||= {};
-
 	let dir = env.CACHE_DIR;
 
-	if (!dir || /^(1|0|true|false)$/.test(dir)) {
-		let pkg = up(options);
+	if (typeof dir === 'undefined' || dir === '1' || dir === '0' || dir === 'true' || dir === 'false') {
+		const pkg = up(options);
 
 		if (dir = pkg && dirname(pkg)) {
-			let mods = join(dir, 'node_modules');
-			let exists = existsSync(mods);
+			const mods = join(dir, 'node_modules');
 
 			// exit cuz exists but not writable
 			// or cuz missing but parent not writable
-			if (!writable(exists ? mods : dir)) return;
+			if (!writable(existsSync(mods) ? mods : dir)) return;
 
 			dir = join(mods, '.cache');
 		}
 	}
 
-	if (dir) {
-		dir = join(dir, name);
-		if (options.create && !existsSync(dir)) {
-			mkdirSync(dir, { recursive: true });
-		}
-		return dir;
+	if (typeof dir === 'string') {
+		const targetDir = join(dir, name);
+		if (options?.create === true && !existsSync(targetDir))
+			mkdirSync(targetDir, { recursive: true });
+
+		return targetDir;
```

Comment:
all stylistic, which is to say, your (lets call it) "fierce" type checking isn't wrong, but it's making no difference in this code's performance and it ignores the runtime conditions that have already taken place.

Also, reassigning to `dir` _is_ better than creating a variable here if you want to hyper optimize. For the same reasons you've removed/inlined variables above.

Reply:
It does because it hints the jit what type you actually want to check instead of doing coercion
Its simply easier to optimize

Reply:
> your "fierce" type checking isn't wrong, but it's making no difference in this code's performance and it ignores the runtime conditions that have already taken place